Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -173,6 +194,12 @@ private bool IsField(FieldInfo field)
                 return true;
             }
 
+            if (!options.IgnoreReadOnlyProperties && 
+                readOnlyProperties.Any(p => field.Name == $"<{p.Name}>k__BackingField"))
```

Comment:
This seems finnicky. I'm not 100% sure what it's for but is there a potentially a better way to do this?

Reply:
I agree - this seems not great, are there alternatives?

Reply:
I wish but I couldn't figure out another way to detect that a given field is a backing field for a given property other than this name comparison.